### ISSUE TITLE
Augmentor for ground truth data

### ISSUE THREAD

**Issue by luciaL** [mdbloice/Augmentor#119]
Hello, I tried the code on the website:
p = Augmentor.Pipeline("/path/to/images")
p.ground_truth("/path/to/ground_truth_images")
p.rotate(probability=1, max_left_rotation=5, max_right_rotation=5)
g = p.keras_generator(batch_size=16)
images, labels = next(g)
Then I fed the generator to keras to train:
model.fit_generator(g,step_per_epoch = 100,epoch =5)
Then I got a error：Cannot feed value of shape(16,1) for Tensor u'conv2d_18_target:0',which has shape(?,?,?,？)

I checked the shape of the pictures in ground_truth file is [160,240].But the labels's shape become [128,1]. I don't know where is the problem with my work.

**Comment by hardik124** [comment]
For some reason, even if ground truth is provided, Keras generator returns an array of 0's as labels

**Comment by 939706171** [comment]
@hardik124 I get the same trouble,and wanna know what the reason it is,and how to solve it.Thank you

**Comment by npielawski** [comment]
I am having the same problem. Looking at the code, the ground truth images can only be exported, otherwise they are dumped. I corrected the code but it isn't backward compatible with respect to the _execute function (it returns an array of images instead of a single image), so I don't know if it would be accepted.

**Comment by mdbloice** [comment]
Hi @luciaL, @hardik124, @npielawski - first off sorry for the delay in replying, I have been away for most of the summer. Are you using TensorFlow? I have not tested Augmentor's generators with TensorFlow, only with Keras so if such issues are occurring with TF I will need to look into it more closely and try to get back to you.

**Comment by tigerdhl** [comment]
i get the same problem when use keras,now have the way to solve it ?
